apps/contributions/tests/test_models.py

```diff
@@ -2088,6 +2088,21 @@ def test_unmarked_abandoned_carts(self, unmarked_abandoned_contributions):
             c.id for c in unmarked_abandoned_contributions
         }
 
+    def test_with_first_payment_date_when_payments_exist(self, monthly_contribution_multiple_payments):
+        contributions = Contribution.objects.with_first_payment_date()
+        first_payment = (
+            Payment.objects.filter(contribution_id=monthly_contribution_multiple_payments.id)
+            .order_by("transaction_time")
+            .first()
+        )
+        assert len(contributions) == 1
+        assert contributions[0].first_payment_date == first_payment.transaction_time
```

Comment:
Can get away with not defining var `first_payment` and also add exclude to ensure that we don't include any payments missing transaction time since that field is nullable.

```suggestion
        contributions = Contribution.objects.with_first_payment_date()
        assert contributions.count() == 1
        assert (con:=contributions.first()).first_payment_date == con.payment_set.exclude(transaction_time__isnull=True).order_by("transaction_time").first().transaction_time
```